### ISSUE TITLE
logs a full of failed to start a background worker

### ISSUE THREAD

**Issue by boskiv** [timescale/helm-charts#620]
Full logs of
```
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [142]: [64f1e8f3.8e-1263] @,app= [01000] WARNING:  failed to launch job 1 "Telemetry Reporter [1]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [141]: [64f1e8f3.8d-8359] @,app= [01000] WARNING:  failed to launch job 1001 "Refresh Continuous Aggregate Policy [1001]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [141]: [64f1e8f3.8d-8360] @,app= [01000] WARNING:  failed to launch job 1000 "Refresh Continuous Aggregate Policy [1000]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [141]: [64f1e8f3.8d-8361] @,app= [01000] WARNING:  failed to launch job 1002 "Refresh Continuous Aggregate Policy [1002]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [141]: [64f1e8f3.8d-8362] @,app= [01000] WARNING:  failed to launch job 1 "Telemetry Reporter [1]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [136]: [64f1e8f3.88-8359] @,app= [01000] WARNING:  failed to launch job 1002 "Refresh Continuous Aggregate Policy [1002]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [141]: [64f1e8f3.8d-8363] @,app= [01000] WARNING:  failed to launch job 1004 "Refresh Continuous Aggregate Policy [1004]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [136]: [64f1e8f3.88-8360] @,app= [01000] WARNING:  failed to launch job 1003 "Refresh Continuous Aggregate Policy [1003]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [141]: [64f1e8f3.8d-8364] @,app= [01000] WARNING:  failed to launch job 1003 "Refresh Continuous Aggregate Policy [1003]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [136]: [64f1e8f3.88-8361] @,app= [01000] WARNING:  failed to launch job 1000 "Refresh Continuous Aggregate Policy [1000]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [136]: [64f1e8f3.88-8362] @,app= [01000] WARNING:  failed to launch job 1001 "Refresh Continuous Aggregate Policy [1001]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [136]: [64f1e8f3.88-8363] @,app= [01000] WARNING:  failed to launch job 1 "Telemetry Reporter [1]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:22 UTC [136]: [64f1e8f3.88-8364] @,app= [01000] WARNING:  failed to launch job 1004 "Refresh Continuous Aggregate Policy [1004]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:23 UTC [140]: [64f1e8f3.8c-2815] @,app= [01000] WARNING:  failed to launch job 1 "Telemetry Reporter [1]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:23 UTC [140]: [64f1e8f3.8c-2816] @,app= [01000] WARNING:  failed to launch job 2 "Error Log Retention Policy [2]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:23 UTC [138]: [64f1e8f3.8a-2815] @,app= [01000] WARNING:  failed to launch job 1 "Telemetry Reporter [1]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:23 UTC [138]: [64f1e8f3.8a-2816] @,app= [01000] WARNING:  failed to launch job 2 "Error Log Retention Policy [2]": failed to start a background worker
timescaledb-1 timescaledb 2023-09-01 14:00:23 UTC [137]: [64f1e8f3.89-1411] @,app= [01000] WARNING
```

**What did you do?**
install timescaledb from helm charts

here is a timescaledb config
```
postgres@timescaledb-2:~$ cat /var/run/postgresql/timescaledb.conf 
max_wal_size = 1GB
min_wal_size = 512MB
shared_preload_libraries = 'timescaledb'
shared_buffers = 802560kB
effective_cache_size = 2351MB
maintenance_work_mem = 401280kB
work_mem = 8025kB
timescaledb.max_background_workers = 16
max_worker_processes = 21
max_parallel_workers_per_gather = 1
max_parallel_workers = 2
wal_buffers = 16MB
default_statistics_target = 500
random_page_cost = 1.1
checkpoint_completion_target = 0.9
max_connections = 50
max_locks_per_transaction = 64
autovacuum_max_workers = 10
autovacuum_naptime = 10
effective_io_concurrency = 256
timescaledb.last_tuned = '2023-09-01T13:45:45Z'
timescaledb.last_tuned_version = '0.14.3'
```

Hereis a db list
```
postgres=# \l
                                                       List of databases
     Name      |     Owner     | Encoding | Collate |  Ctype  | ICU Locale | Locale Provider |        Access privileges        
---------------+---------------+----------+---------+---------+------------+-----------------+---------------------------------
 core          | core          | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | 
 core_api      | core_api      | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | 
 core_storage  | core_storage  | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | 
 postgres      | postgres      | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | 
 sf_go_core    | sf_go_core    | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =Tc/sf_go_core                 +
               |               |          |         |         |            |                 | sf_go_core=CTc/sf_go_core
 sf_go_funding | sf_go_funding | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =Tc/sf_go_funding              +
               |               |          |         |         |            |                 | sf_go_funding=CTc/sf_go_funding
 sf_go_news    | sf_go_news    | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =Tc/sf_go_news                 +
               |               |          |         |         |            |                 | sf_go_news=CTc/sf_go_news
 sf_go_oi      | sf_go_oi      | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =Tc/sf_go_oi                   +
               |               |          |         |         |            |                 | sf_go_oi=CTc/sf_go_oi
 template0     | postgres      | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =c/postgres                    +
               |               |          |         |         |            |                 | postgres=CTc/postgres
 template1     | postgres      | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =c/postgres                    +
               |               |          |         |         |            |                 | postgres=CTc/postgres
(10 rows)
```

**Did you expect to see some different?**

**Environment**
k8s
* Which helm chart and what version are you using?
0.33.1
* What is in your `values.yaml` ?
```
  values:
    clusterName: "sf-cluster"
    image:
      tag: pg15.4-ts2.11.2-all
    persistentVolumes:
      data:
        size: 100Gi
      wal:
        size: 10Gi
    pgBouncer:
      enabled: true
      config:
        ignore_startup_parameters: extra_float_digits
        server_reset_query: DISCARD ALL
        max_client_conn: 500
        default_pool_size: 12
        pool_mode: transaction
    prometheus:
      enabled: true
    tolerations:
      - key: "node.k8s/role"
        operator: "Equal"
        value: "timescaledb"
        effect: "NoSchedule"
```

* Kubernetes version information:

```
➜  ~ kubectl version
Client Version: v1.28.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.27.4-eks-2d98532
```

    

* Kubernetes cluster kind:

eks

**Anything else we need to know?**:

**Comment by boskiv** [comment]
When I do 
```
select * from pg_settings where name = 'max_worker_processes';
```

I've got 8

**Comment by boskiv** [comment]
The only way I succeed is overwrite  values in
```
patroni:
      bootstrap:
        dcs:
          postgresql:
            parameters:
```
But it's not a right way. If I upgrade my nodes i have to update values again.

**Comment by boskiv** [comment]
Hi, @theblinkingusb . Thanks for reply. Yes, I have feelings that parameters from /var/run/postgresql/timescaledb.conf are not applying. Only way I can fix it, is specify in helm values.

**Comment by theblinkingusb** [comment]
Yeah, I also ended up doing the the same thing. I don't yet have a solution but my comments thus  far- 

I did do a fresh install of 15.4 with 2.11 today - my timescale.conf did not have a timescaledb.max_background_workers in it (and it was reporting a recent tstune date) - but I was getting the same error you were with the background worker failed to launch. I was also getting this error with the 0.33.1 chart version (pg 14), I (wrongly) had not noticed it.

I tuned the params within the helm chart (`timescaledb.max_background_workers, max_worker_processes, max_parallel_workers` under `patroni....params`, using settings based on the aforementioned docs) and redeployed - which silenced the errors (and reduced cpu/mem usage).

Interestingly (and likely expectedly to some lol) those settings passed into the data dir postgres.conf, not timescale.conf. Actually my timescale.conf settings remained the same, and the pg_settings reflected the changes from the helm settings.

All that said - I think this behavior makes sense in my case because I chose to edit the resource requests and limits in my helm chart. However @boskiv - looks like you kept those as default - so I could agree that from a helm perspective those settings should be defaulted such that it doesn't cause any error when deploying with defaults. But I don't know enough yet about how ts-tune interplays with the .conf settings, the interplay between ts-tune and k8s defaults, and the interplay between pg conf and timescale conf. Also we are both deploying with newer images than the chart specifies, but based on the release notes between the pg/ts versions it doesn't look like that has an affect on this behavior.

**Comment by wollud1969** [comment]
I want to come back to this question, since I've the same problem: the log full of "failed to start a background worker" messages. 
I've updated my `values.yml` to contain:

```
patroni:
  bootstrap:
    dcs:
      postgresql:
        parameters:
          max_worker_processes: 32
          max_parallel_workers: 16
          timescaledb.max_background_workers: 16

```

and after a helm upgrade cycle I found those settings in the patroni configmap and also in the `/etc/timescaledb/patroni.yml` in the pods. But even after restarting the pods, the changed settings are neither in the `postmaster.opts` (also not after deleting the file and restarting the pod again as proposed here [https://github.com/timescale/helm-charts/issues/504](https://github.com/timescale/helm-charts/issues/504)) nor in the `postgresql.conf`.
 
At github/zalando/patroni ([https://github.com/zalando/patroni/issues/2315](https://github.com/zalando/patroni/issues/2315)) I read, the the bootstrap configuration will only be applied when a postgres cluster is created for the first time. Does that apply also here? And if so, what is the proposed way to change those parameters afterwards?

Thanks and Cheers, Wolfgang
